### PR TITLE
382: List commits that have been pushed in integration message

### PR DESCRIPTION
**Pull request by rwestberg**
Hi all,

Please review this change that lists the commits that have been pushed since the PR was created (up to 10).

Best regards,
Robin

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [SKARA-382](https://bugs.openjdk.java.net/browse/SKARA-382): List commits that have been pushed in integration message

### Reviewers
 * Erik Helin ([ehelin](@edvbld) - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/skara pull/602/head:pull/602`
`$ git checkout pull/602`

### CODE REVIEW

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java [hunk 1]
```diff
@@ -442,7 +442,15 @@ private String getMergeReadyComment(String commitMessage, List<Review> reviews)
             }
             message.append("pushed to the `");
             message.append(pr.targetRef());
-            message.append("` branch. ");
+            message.append("` branch:\n\n");
+            divergingCommits.stream()
+                            .limit(10)
```
Comment (edvbld) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java:447]: Will this be the most recent or the oldest 10 diverging commits?

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java [hunk 2]
```diff
@@ -442,7 +442,15 @@ private String getMergeReadyComment(String commitMessage, List<Review> reviews)
...
+                            .forEach(c -> message.append(" * ").append(c.hash().hex()).append(": ").append(c.message().get(0)).append("\n"));
+            if (divergingCommits.size() > 10) {
+                message.append(" * ... full list: ").append(pr.repository().webUrl(baseHash.hex(), pr.targetRef())).append("\n");
```
Comment (edvbld) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java:450]: Can we say something like `* ... and 5 more: `?

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java (hunk 1 above)
Comment (rwestberg) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java:447]: This should give the 10 most recent.

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java (hunk 2 above)
Comment (rwestberg) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java:450]: Sure, but the link itself should show a full list, right?

**Comment by rwestberg** [comment]
Thanks for reviewing!

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java (hunk 1 above)
Comment (edvbld) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java:447]: Sounds good :+1:

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java (hunk 2 above)
Comment (edvbld) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java:450]: Yeah, I think that is the best

**Pull request closed without merging**